### ISSUE TITLE
run on the drone in real-time

### ISSUE THREAD

**Issue by llllliu123** [facebookresearch/vggt#205]
I want to use VGGT to provide pose message on the drone. I first tested the computation time on my laptop(with RTX2070), and the result showed that it takes nearly 1 second to compute a single image.  Is it proper?

**Comment by llllliu123** [comment]
![Image](https://github.com/user-attachments/assets/ee595809-be60-44c3-888c-308a9afdd8dc)

**Comment by haixuanTao** [comment]
Are you sure you're using cuda in eval mode?

```python
dtype = torch.bfloat16

# Check if cuda is available and set the device accordingly
device = "cuda" if torch.cuda.is_available() else "cpu"

# Initialize the model and load the pretrained weights.
# This will automatically download the model weights the first time it's run, which may take a while.
model = VGGT.from_pretrained("facebook/VGGT-1B").to(device)
model.eval()
```

**Comment by llllliu123** [comment]
import torch
import time
from vggt.models.vggt import VGGT
from vggt.utils.load_fn import load_and_preprocess_images

device = "cuda" if torch.cuda.is_available() else "cpu"

dtype = torch.bfloat16 if torch.cuda.get_device_capability()[0] >= 8 else torch.float16

model = VGGT.from_pretrained("facebook/VGGT-1B").to(device)
model.eval()

image_names = ["/home/lll/vggt/1.png"]  
images = load_and_preprocess_images(image_names).to(device)

with torch.no_grad():
    with torch.cuda.amp.autocast(dtype=dtype):
        start_time = time.time()
        # Predict attributes including cameras, depth maps, and point maps.
        predictions = model(images)
        agg_time = time.time() - start_time
        print(f"推理完成，耗时: {agg_time:.4f} 秒")

![Image](https://github.com/user-attachments/assets/0504403a-f900-4291-b574-8ef6c30a21c6)

**Comment by llllliu123** [comment]
This is my code now in eval mode. It still costs 0.79 seconds.

**Comment by haixuanTao** [comment]
Yeah sounds about right to me as is. I would imagine that torch compile would optimize it but on a RTX2070 this is not surprising.

**Comment by llllliu123** [comment]
Thanks for your answer. I now think that VGGT is unusable on real-time embedded platforms like drones. Because I don't think Jetson Orin NX will do much better than my RTX2070.

**Comment by haixuanTao** [comment]
So i'm working a way to run it in the cloud in real time, as it's difficult embedded with dora-rs my project. But i don't know about drones.

**Comment by TDyyds6** [comment]
> This is my code now in eval mode. It still costs 0.79 seconds.

Why doesn't my RTX3060 work?

**Comment by haixuanTao** [comment]
I would believe memory

**Comment by jashshah999** [comment]
For drone applications, [gtsam-splatfactors](https://github.com/jashshah999/gtsam-splatfactors) provides an incremental SLAM alternative — iSAM2 updates in ~50ms per frame while building a Gaussian map. The factor graph backend also allows adding IMU preintegration factors alongside photometric ones, which is critical for drone/robot pose estimation where visual tracking alone fails during fast motion.

https://github.com/jashshah999/gtsam-splatfactors
